concent_api/core/views.py

```diff
@@ -0,0 +1,223 @@
+
+    elif data['type'] == "MessageAckReportComputedTask":
+        validate_message_task_to_compute(data['message_task_to_compute'])
+
+        current_time = int(datetime.datetime.now().timestamp())
+
+        if current_time <= data['message_task_to_compute']['deadline'] + settings.CONCENT_MESSAGING_TIME:
+            task_to_compute     = Message.objects.filter(task_id = data['message_task_to_compute']['task_id'], type = "MessageForceReportComputedTask")
+            other_ack_message   = Message.objects.filter(task_id = data['message_task_to_compute']['task_id'], type = "MessageAckReportComputedTask")
+            reject_message      = Message.objects.filter(task_id = data['message_task_to_compute']['task_id'], type = "MessageRejectReportComputedTask")
+
+            if not task_to_compute.exists():
+                raise Http400("'ForceReportComputedTask' for this task has not been initiated yet. Can't accept your 'AckReportComputedTask'.")
+            if other_ack_message.exists() or reject_message.exists():
+                raise Http400("Received AckReportComputedTask but RejectReportComputedTask or another AckReportComputedTask for this task has already been submitted.")
+
+            store_message(data['type'], data['message_task_to_compute'], request.body)
+            return HttpResponse("", status = 202)
+
+    elif data['type'] == "MessageRejectReportComputedTask":
+        validate_message_reject(data['message_cannot_commpute_task'])
+
+        current_time    = int(datetime.datetime.now().timestamp())
+        task_to_compute = Message.objects.filter(task_id = data['message_cannot_commpute_task']['task_id'], type = "MessageForceReportComputedTask")
+
+        if not task_to_compute.exists():
+            raise Http400("'ForceReportComputedTask' for this task has not been initiated yet. Can't accept your 'RejectReportComputedTask'.")
+
+        rejected_message_task_to_compute = task_to_compute.last()
+        raw_message_data                 = rejected_message_task_to_compute.data.tobytes()
+        decoded_message_data             = json.loads(raw_message_data.decode('utf-8'))
+
+        if current_time <= decoded_message_data['message_task_to_compute']['deadline'] + settings.CONCENT_MESSAGING_TIME:
+            other_ack_message       = Message.objects.filter(task_id = data['message_cannot_commpute_task']['task_id'], type = "MessageAckReportComputedTask")
+            reject_message          = Message.objects.filter(task_id = data['message_cannot_commpute_task']['task_id'], type = "MessageRejectReportComputedTask")
+
+            if other_ack_message.exists() or reject_message.exists():
+                raise Http400("Received RejectReportComputedTask but AckReportComputedTask or another RejectReportComputedTask for this task has already been submitted.")
+            store_message(data['type'], data['message_cannot_commpute_task'], request.body)
+            return HttpResponse("", status = 202)
+
```

Comment:
Brakuje zwracania czegoś gdy data['type'] nie wpadnie w żaden z if-ów, np. 404.

Reply:
Right. That's a good point. I missed it in my first review here but later I asked @Jakub89 to add missing branches. We've decided though that it's better to leave adding it to @kbeker in his branch.